src/test/java/uk/gov/hmcts/pdda/business/services/formatting/FirmListXmlMergeUtilsNodesTest.java

```diff
@@ -0,0 +1,156 @@
+
+import static org.junit.jupiter.api.Assertions.assertTrue;
+
+/**
+ * <p>
+ * Title: FirmListXMLMergeUtils Second Test Class.
+ * </p>
+ * <p>
+ * Description:
+ * </p>
+ * <p>
+ * Copyright: Copyright (c) 2024
+ * </p>
+ * <p>
+ * Company: CGI
+ * </p>
+ * 
+ * @author Luke Gittins
+ */
+@ExtendWith(EasyMockExtension.class)
+class FirmListXmlMergeUtilsNodesTest {
+
+    private static final String TRUE = "Result is True";
+
+    @TestSubject
+    private static FirmListXmlMergeUtils classUnderTest;
+
+    @BeforeAll
+    public static void setUp() throws Exception {
+        classUnderTest = new FirmListXmlMergeUtils();
+    }
+
+    @AfterAll
+    public static void tearDown() throws Exception {
+        // Do nothing
+    }
+
+    @Test
+    void testMergeNodeCombinations()
+        throws SAXException, IOException, ParserConfigurationException {
+        List<String> documentXmls = new ArrayList<>();
+        List<String> documentXmlsToMerge = new ArrayList<>();
+        boolean result = true;
+        // Insert before Data
+        documentXmls.add(
+            "<cs:CourtList><cs:CourtHouse>TestCourtHouse</cs:CourtHouse><cs:FirmList>"
+            + "<cs:HearingDate>01/01/2024</cs:HearingDate></cs:FirmList></cs:CourtList>");
+        documentXmlsToMerge.add(
+            "<cs:FirmList><cs:ReserveList><cs:HearingDate>01/01/2024</cs:HearingDate>"
+            + "</cs:ReserveList></cs:FirmList>");
+        // Second Node Reserved Data
+        documentXmls.add("<cs:FirmList></cs:FirmList>");
+        documentXmlsToMerge.add("<cs:ReserveList></cs:ReserveList>");
+        // First Node Reserved Data
+        documentXmls.add("<cs:ReserveList></cs:ReserveList>");
+        documentXmlsToMerge.add("<cs:ReserveList></cs:ReserveList>");
+
+        for (int i = 0; i < documentXmls.size(); i++) {
+            testMergeNode(documentXmls.get(i), documentXmlsToMerge.get(i));
+            assertTrue(result, TRUE);
+        }
+    }
+
+    void testMergeNode(String documentXml, String documentXmlToMerge)
+        throws SAXException, IOException, ParserConfigurationException {
+        
+        Document baseDocument = getDummyDoc(documentXml);
+        Node node = baseDocument.getDocumentElement();
+        Node nodeToMerge = getDummyDoc(documentXmlToMerge).getDocumentElement();
+
+        classUnderTest.mergeNode(baseDocument, node, nodeToMerge);
+    }
+
+    @Test
+    void testMergeNodeEmptyParentNode()
+        throws SAXException, IOException, ParserConfigurationException {
+        String documentXml =
+            "<cs:FirmList><cs:ReserveList></cs:ReserveList><cs:CourtList></cs:CourtList></cs:FirmList>";
+        String documentXmlToMerge =
+            "<cs:FirmList><cs:ReserveList><cs:HearingDate>01/01/2024</cs:HearingDate></cs:ReserveList><cs:CourtList>"
+                + "<cs:CourtHouse>SNARESBROOK</cs:CourtHouse></cs:CourtList></cs:FirmList>";
+        boolean result = true;
+
+        Document baseDocument = getDummyDoc(documentXml);

```

Comment:
The setup from here below is repeated in the below method as well, please extract this

Reply:
To centralise this would be more work than what's already there. Either using global variables and making several methods to set these values or creating a whole new class and setting these values as attributes then referencing it in the test are 2 ways I can think but what's there already seems the simplest route.

Reply:
cool